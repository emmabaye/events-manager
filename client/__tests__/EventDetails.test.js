import { BrowserRouter } from 'react-router-dom';
import ConnectedEventDetails, { EventDetails } from '../components/EventDetails.jsx';

describe('EventDetails Component', () => {
  const initialState = {
    eventReducer: {
      status: "Error",
      message: "",
      event: {
        id: '1',
        title: 'Seminar',
        description: "",
        location: "",
        time: "5:00",
        date: '2020-01-01',
        image: "",
        Center: []
      }
    }
  };

  const store = mockStore(initialState);

  global.localStorage = {
    getItem: (str) => str
  };

  const props = {
    dispatch: () => {},
    centerId: "1",
    event: initialState.eventReducer.event,
    match: {
      params: {
        id: '1'
      }
    },
  };

  describe('EventDetails form  should render self', () => {
    it('it should render for dumb component', () => {
      const wrapper = shallow(<BrowserRouter><EventDetails {...props} /></BrowserRouter>);
      expect(wrapper).toMatchSnapshot();
    });

    it('it should render connected component', () => {
      const wrapper = mount(<BrowserRouter><Provider store={store}><ConnectedEventDetails match={props.match}/></Provider></BrowserRouter>);
      expect(wrapper.length).toEqual(1);
    });
  });
});
