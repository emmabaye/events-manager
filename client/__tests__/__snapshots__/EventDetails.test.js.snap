// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventDetails Component EventDetails form  should render self it should render for dumb component 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "blank",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <EventDetails
    centerId="1"
    dispatch={[Function]}
    event={
      Object {
        "Center": Array [],
        "date": "2020-01-01",
        "description": "",
        "id": "1",
        "image": "",
        "location": "",
        "time": "5:00",
        "title": "Seminar",
      }
    }
    match={
      Object {
        "params": Object {
          "id": "1",
        },
      }
    }
  />
</Router>
`;
